crates/sui-core/src/authority_server.rs

```diff
@@ -999,20 +999,18 @@ macro_rules! handle_with_decoration {
                 let do_header_parse = |op: &MetadataValue<Ascii>| {
                     match op.to_str() {
                         Ok(header_val) => {
-                            match header_val.parse::<SocketAddr>() {
-                                Ok(socket_addr) => Some(socket_addr.ip()),
-                                Err(err) => {
+                            header_val.parse::<IpAddr>().ok().or_else(|| {
+                                header_val.parse::<SocketAddr>().ok().map(|socket_addr| socket_addr.ip()).or_else(|| {
```

Comment:
might be worth pulling this code out into a function, both for clarity and for re-use with axum_router